lib/utils.js

```diff
@@ -375,9 +375,29 @@ function parseSettings(urlStr) {
  *
  * @param {Object} target The target settings object
  * @param {Object} src The source settings object
+ * @param {Boolean} replaceTarget Should the src object replace the target value if it exists
  * @returns {Object} The merged settings object
  */
-function mergeSettings(target, src) {
+function mergeSettings(target, src, replaceTarget) { // NOTE: src is subclassSettings
+  const mergePolicy = {
+    description: {replace: true}, // string or array
+    options: {innerReplace: true}, // object
+    properties: {innerReplace: true}, // object
+    hidden: {replace: false}, // array
+    protected: {replace: false}, // array
+    indexes: {innerReplace: true}, // object
+    methods: {innerReplace: true}, // object
+    mixins: {innerReplace: true}, // object
+    // validations: {innerReplace: true}, // object // NOTE: not implemented
+    relations: {innerReplace: true},
+    scope: {replace: true}, // object
+    scopes: {innerReplace: true}, // object
+    acls: {replace: false}, // array
```

Comment:
From merging perspective, `{replace: false}` will instruct LoopBack to combine ACL rules from both base and sub model. I'm wondering if we should introduce a way to generalize the idea of `ranking by inheritance hierarchy`. Really, our inheritance is much closer to mixin based composition.